### PR TITLE
CBG-1323: Include xattr in _raw

### PR DESCRIPTION
**Pull request by JRascagneres**

### CODE REVIEW

File: rest/admin_api.go [hunk 1]
```diff
@@ -403,6 +403,17 @@ func (h *handler) handleGetRawDoc() error {
...
+	if h.db.Options.UserXattrKey != "" {
+		rawUserXattrVal := doc.RawUserXattr
+		if len(rawUserXattrVal) == 0 {
+			rawUserXattrVal = []byte("null")
+		}
+		rawBytes, err = base.InjectJSONPropertiesFromBytes(rawBytes, base.KVPairBytes{Key: h.db.Options.UserXattrKey, Val: rawUserXattrVal})
+		if err != nil {
+			return err
+		}
+	}
```
Comment (bbrks) [inline rest/admin_api.go]: This seems a bit mis-representative, given that the xattr in the sync function is nested under "meta", and this wouldn't.

Also what would happen if the doc already contains a property matching the xattr key (e.g. `"channels"`)?

**Review by bbrks — CHANGES_REQUESTED** [review]

File: rest/admin_api.go (hunk 1 above)
Comment (JRascagneres) [reply rest/admin_api.go]: Thats a fair point... Similarly a user could have a 'meta' field in their doc though.

File: rest/admin_api.go [hunk 2]
```diff
@@ -403,6 +403,21 @@ func (h *handler) handleGetRawDoc() error {
...
+
+	metaMapBytes, err := base.JSONMarshal(metaMap)
+	if err != nil {
+		return err
+	}
+
+	rawBytes, err = base.InjectJSONPropertiesFromBytes(rawBytes, base.KVPairBytes{Key: "_meta", Val: metaMapBytes})
+	if err != nil {
+		return err
+	}
```
Comment (bbrks) [inline rest/admin_api.go]: Nitpick: There's a variant of this util that does the marshalling for you. But don't bother changing unless there's other stuff to do.

```suggestion
	rawBytes, err = base.InjectJSONProperties(rawBytes, base.KVPair{Key: "_meta", Val: metaMap})
	if err != nil {
		return err
	}
```

File: rest/admin_api.go [hunk 3]
```diff
@@ -403,6 +403,21 @@ func (h *handler) handleGetRawDoc() error {
 		return err
 	}
 
+	metaMap, err := doc.GetMetaMap(h.db.Options.UserXattrKey)
```
Comment (adamcfraser) [inline rest/admin_api.go]: Don't show meta unless userXattrKey is set.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

**Review by bbrks — APPROVED** [review]

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**